src/runtime-pmi2.c

```diff
@@ -95,14 +96,9 @@ decode(const char *inval, void *outval, int outvallen)
 int
 shmem_runtime_init(void)
 {
-    int spawned, appnum, my_node;
-    int initialized;
+    int spawned, appnum;
 
-    if (PMI2_SUCCESS != PMI2_Initialized()) {
-        return 1;
-    }
-
-    if (!initialized) {
+    if (!PMI2_Initialized()) {
```

Comment:
Double-checked the MPICH PMI2 header.  This change looks correct:

```
/*@
  PMI2_Initialized - check if PMI has been initialized

  Return values:
  Non-zero if PMI2_Initialize has been called successfully, zero otherwise.
@*/
```